server_addon/maya/client/ayon_maya/plugins/load/load_ornatrix_cache.py

```diff
@@ -0,0 +1,164 @@
+        group_node = cmds.group(nodes, name=group_name)
+        project_name = context["project"]["name"]
+
+        settings = get_project_settings(project_name)
+        color = get_load_color_for_product_type(product_type, settings)
+        if color is not None:
+            red, green, blue = color
+            cmds.setAttr(group_node + ".useOutlinerColor", 1)
+            cmds.setAttr(group_node + ".outlinerColor", red, green, blue)
+
+        nodes.append(group_node)
+
+        self[:] = nodes
+
+        return containerise(
+            name=name,
+            namespace=namespace,
+            nodes=nodes,
+            context=context,
+            loader=self.__class__.__name__
+        )
+
+    def remove(self, container):
+
+        namespace = container["namespace"]
+        nodes = container["nodes"]
+
+        self.log.info("Removing '%s' from Maya.." % container["name"])
+
+        nodes = cmds.ls(nodes, long=True)
+
+        try:
+            cmds.delete(nodes)
+        except ValueError:
+            # Already implicitly deleted by Maya upon removing reference
+            pass
+
+        cmds.namespace(removeNamespace=namespace, deleteNamespaceContent=True)
+
+    def update(self, container, context):
+        repre_entity = context["representation"]
+        nodes = container["nodes"]
+
+        path = get_representation_path(repre_entity)
+        for node in nodes:
+            if cmds.ls(node, type="HairFromGuidesNode"):
+                cmds.setAttr(f"{node}.cacheFilePath", path)
+
+
+    def switch(self, container, context):
+        self.update(container, context)
+
+    # helper functions
+    def create_namespace(self, folder_name):
+        """Create a unique namespace
+        Args:
+            asset (dict): asset information
+
+        """
+
+        asset_name = "{}_".format(folder_name)
+        prefix = "_" if asset_name[0].isdigit() else ""
+        namespace = lib.unique_namespace(
+            asset_name,
+            prefix=prefix,
+            suffix="_"
+        )
+
+        return namespace
+
+    def create_node(self, namespace, filepath, node_settings):
+        """Use the cachesettings to create a shape node which
+        connects to HairFromGuidesNode with abc file cache.
+
+        Args:
+            namespace (str): namespace
+            filepath (str): filepath
+            node_settings (dict): node settings
+
+        Returns:
+            _type_: _description_
+        """
+        nodes = []
+        orig_guide_name = node_settings["name"]
+        orig_shape_name = node_settings["shape"]["name"]
+        mesh_shape_name = "{}:{}".format(namespace, orig_shape_name)
+        guide_name = "{}:{}".format(namespace, orig_guide_name)
+        mesh_shape_node = cmds.ls(guide_name, type="mesh")
+        if not mesh_shape_node:
+            mesh_shape_node = cmds.createNode("mesh", name=guide_name)
+        hair_guide_node = cmds.ls(guide_name, type="HairFromGuidesNode")
```

Comment:
This would be a `List[str]` which seems odd with the variable name, bug?
Also, the usage below with `nodes.append(hair_guide_node)` then seems wrong.

You might be better of just doing:
```python
hair_guide_node = guide_name
if not cmds.objExists(hair_guide_node):
    hair_guide_node = cmds.createNode('mesh", name=guide_name)
```

Also it feels wrong that we're looking for a guide name AND a shape name, but then if it does not exist create the _shape_ with the guide name? Are we mixing up things here by accident?

Note that if you create a `mesh` type node in Maya that it will ALWAYS also create a parent transform automatically if you have not named it or supplied a parent so this seems like you're not taking the full control you actually want?

Reply:
> Note that if you create a `mesh` type node in Maya that it will ALWAYS also create a parent transform automatically if you have not named it or supplied a parent so this seems like you're not taking the full control you actually want?

It is something challenging for loading the cached abc in the guide node as it is connected to the shape node of the connection. And the Operator stack requires you to connecting these two together in order to show up the guide node (and the shape node) in the stack editor. Would be great if you can suggest solution for this case.

Reply:
> Could share the Node graph of what it looks like in Maya when the connections are correct (and e.g. this is currently loaded?) Like some screenshots.

already DM you in discord, it's a clip